x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/analysis/analyzer/Verifier.java

```diff
@@ -406,7 +411,7 @@ private static boolean checkGroupByHavingHasOnlyAggs(Expression e, Set<Expressio
 
         // resolve FunctionAttribute to backing functions
         if (e instanceof ReferenceAttribute) {
-            e = attributeRefs.get(e);
+            e = attributeRefs.resolve(e, null);
```

Comment:
The null variant is used enough times to make a `resolve` method, in addition to `resolveOrDefault` on the main class similar to `get` and `getOrDefault` (even if it's just syntactic sugar).

Reply:
✓ Done.